service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/ControlledAzureVmResource.java

```diff
@@ -56,10 +63,12 @@ public ControlledAzureVmResource(
       @JsonProperty("vmName") String vmName,
       @JsonProperty("region") String region,
       @JsonProperty("vmSize") String vmSize,
-      @JsonProperty("vmImageUri") String vmImageUri,
+      @JsonProperty("vmImage") ApiAzureVmImage vmImage,
```

Comment:
There are a few considerations, so let me clarify a bit.

On GCP, we do not store attributes of buckets, because the application - notebook, workflow, UI - can call the cloud API directly to learn those features. On Azure, since the current implementation does not support cloud transparency, 
these attributes cannot be read directly. If we need to support reading them, then maybe we should keep them in WSM. I do not want WSM to be responsible for making cloud platform accesses to read these. But maybe since this is only for Leo, then Leo can read what it needs without WSM being involved.

Either way, I do not want to setup the situation where changing the API will cause compatibility issues with what is stored in the database. So at least the `ControlledAzureVmAttributes` should not be making any reference to the API classes. I would prefer that the `ControlledAzureVmResource` also not make any reference.

@Qi77Qi I'd be interested in your thoughts.

Reply:
yea we don't need to store these in DB...we will store them in Leonardo anyway just for informational purpose I think

Reply:
My assumption about workflow was a little bit wrong. Removed it from there.